gateway/api/views.py

```diff
@@ -890,3 +904,32 @@ def retrieve(self, request, pk=None):
                 instance, context={"user": user}
             )
         return Response(serializer.data)
+
+    @action(methods=["GET"], detail=False)
+    def get_by_title(self, request):
+        """Get a specific program in the catalog:"""
+        tracer = trace.get_tracer("gateway.tracer")
+        ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
+        with tracer.start_as_current_span("gateway.catalog.retrieve", context=ctx):
+            title = self.request.query_params.get("title")
+            provider_name = self.request.query_params.get("provider")
+            if not title or not provider:
+                return Response(
+                    {"message": "Qiskit Function not found."},
+                    status=status.HTTP_404_NOT_FOUND,
```

Comment:
Instead to return a 404 how we expected that they pass both params I would use a 400 BAD_REQUEST and clarify that both filters are needed in the message.